cEP-0026.md

```diff
@@ -0,0 +1,189 @@
+  (the parent directory's path; in case of project's root it will be `None`)
+- There will be a separate dict (called directory-tracker) containing directory
+  paths as keys and number of files and subdirs as a list of values for e.g.
+  ```
+      { Directory.path: [Directory.files, Directory.subdirs] }
+  ```
+- First all the unchanged files will be deleted from the cache whenever a file
+  is deleted from the cache we will look up its parent directory using
+  `FileFactory.parent` and decrease the count of the number of files
+  (`Directory.files` in the directory-tracker dict) corresponding to that path
+  by one.
+- Once all the files have been deleted from the cache coala will move on to the
+  directories in our dict. coala will look for directories with
+  `Directory.subdirs` = 0 (starting from the bottommost level of the directory
+  tree) if the `Directory.files` associated with such paths is 0 (i.e. all the
+  files contained inside that directory have been uncached) then the
+  `Directory` object corresponding to this path will be deleted from the cache.
+- As soon as a Directory object is uncached the value of the
+  `Directory.subdirs` of their parent directory will also be decreased by 1.
+- This process will continue for all the directories and the paths having both
+  the values (`Directory.files` and `Directory.subdirs`) as zero will have
+  their corresponding `Directory` objects deleted from the cache.
+
+### Cache control flags
+
+This feature enhancement focusses both on performance and usability. The aim
+is to provide different caching modes through cache control flags
+to provide the user the flexibility to tweak the caching mechanism according to
+their project's needs. The following cache flags can be implemented to
+provide different caching modes:
+
+- `--cache-protocol`: This will accept three different arguments, `none` if
+   the user wants to run coala without caching enabled, `primitive` to store
+   all the cache entries without removing the unchanged ones on subsequent runs
+   (memory wise quite inefficient but this will prove to be useful in project's
+   with small incremental builds that need fast linting), `untrack` the
+   opposite of the `primitive` mode which will stop tracking unchanged files
+   and directories on subsequent coala runs and `lru` (last recently used) in
+   which cached items only persist until the next run (a count parameter can
+   also be used to know when to discard an unused cached item).
+- `--clear-cache`: Similar to `--flush-cache`.
+- `--cache-compression`: This will accept the following arguments: `none`
+   for compression to be disabled and `lzma` or `gzip` to compress the cache
+   using these modules.
+- `cache-optimization`: For faster cache loading we can make use of this flag
+   which will utilize `pickletools.optimize` for faster cache loading.
+- `export-cache` or `import-cache`: To transfer cache across different
```

Comment:
I would add "API":
> ...:  API to transfer cache ...

As it's yet to be designed how coala will store the cache ;)